tests/python_package_test/test_sklearn.py

```diff
@@ -609,21 +654,21 @@ def test_pandas_categorical():
     X[cat_cols_actual] = X[cat_cols_actual].astype('category')
     X_test[cat_cols_actual] = X_test[cat_cols_actual].astype('category')
     cat_values = [X[col].cat.categories.tolist() for col in cat_cols_to_store]
-    gbm0 = lgb.sklearn.LGBMClassifier(n_estimators=10).fit(X, y)
+    gbm0 = lgb.sklearn.LGBMClassifier(n_estimators=10, random_state=42).fit(X, y)
```

Comment:
Why was this change necessary?

Reply:
Thanks for spotting this, it comes from a previous implementation where early stopping was true by default, now it is 'auto' by default. This test does not use early stopping indeed.

Reply:
Ok thank you. Please, as you continue on this, check the diff and be confident that every change you're proposing is necessary and can be justified. The more you can eliminate these "what is this? ... oh nevermind we don't need it any more" rounds of reviews, the faster this will get to a state that we feel confident merging.

Reply:
Sure, thanks for the suggestion.